### PR TITLE
TreeSet.addAll(It) fixed

### PR DESCRIPTION
**Pull request by ruslansennov**

### CODE REVIEW

File: src/main/java/javaslang/collection/TreeSet.java [hunk 1]
```diff
@@ -409,18 +409,14 @@
     @Override
     public TreeSet<T> addAll(java.lang.Iterable<? extends T> elements) {
         Objects.requireNonNull(elements, "elements is null");
-        if (isEmpty() && elements instanceof TreeSet) {
-            return (TreeSet<T>) elements;
```
Comment (ruslansennov) [inline src/main/java/javaslang/collection/TreeSet.java]: it was wrong in case of <code>this.comparator != elements.comparator</code>

**Comment by danieldietrich** [comment]
You are right! Many thanks

**Comment by danieldietrich** [comment]
Btw, for your planning, I will push RC1 to maven central on 15. Nov.

I need to do s.th. different than coding in between, s.th. healthy :-)

**Comment by ruslansennov** [comment]
gogogo

**Comment by danieldietrich** [comment]
yes, will do.

**Pull request merged**